Move JS API out of debug ID proposal

lforst opened the pull request:
- No longer suggest a JS API in the main part of the proposal (we still discuss possible options in an appendix to highlight the value for post-hoc debugging)
- Be more specific about JS APIs (ref https://github.com/tc39/ecma426/issues/146)
- Remove unnecessary details from symbol server section (we don't care about the storage mechanism)

cc @szuend

szuend reviewed (approved):
Thanks!

I think it's also worth to call out an additional use-case for debugIDs, which is source map caching in debuggers. Source maps keyed on URLs can't be cached without loading them + hashing the content. With debugIDs, browser dev tools and IDEs can decide upfront whether they need to load + parse a source map or not.

lforst commented:
Great point. Thank you! I added it to the use-cases section.

I also added a short explainer that whether or not a debug ID is deterministic is encoded in the version bits of the UUID, which is an important bit (pun not intended) of information since devtools and debuggers may probably only want to apply caching mechanisms when the debug ID is deterministic.

lforst commented:
@szuend @nicolo-ribaudo I think this would be fine to merge if any of you want to do it. :) I don't have merge permissions myself.
